Mathlib/Data/Matroid/Loop.lean

```diff
@@ -734,4 +745,131 @@ lemma ext_indep_disjoint_loops_coloops {M₁ M₂ : Matroid α} (hE : M₁.E = M
+  obtain ⟨R, hR, rfl⟩ := hR
+  rw [loopless_iff, restrict_loops_eq hR, M.loops_eq_empty, empty_inter]
+
+instance {M : Matroid α} [Matroid.Nonempty M] [Loopless M] : RankPos M :=
+  M.ground_nonempty.elim fun _ he ↦ (toIsNonloop he).rankPos
+
+@[simp] lemma loopyOn_isLoopless_iff {E : Set α} : Loopless (loopyOn E) ↔ E = ∅ := by
+  simp [loopless_iff_forall_not_isLoop, eq_empty_iff_forall_not_mem]
+
+/-- The loopless matroid obtained from `M` by deleting all its loops. -/
+def removeLoops (M : Matroid α) : Matroid α := M ↾ {e | M.IsNonloop e}
+
+lemma removeLoops_eq_restrict (M : Matroid α) : M.removeLoops = M ↾ {e | M.IsNonloop e} := rfl
+
+lemma removeLoops_ground_eq (M : Matroid α) : M.removeLoops.E = {e | M.IsNonloop e} := rfl
+
+instance removeLoops_isLoopless (M : Matroid α) : Loopless M.removeLoops := by
+  simp [loopless_iff_forall_isNonloop, removeLoops]
+
+@[simp]
+lemma removeLoops_eq_self (M : Matroid α) [Loopless M] : M.removeLoops = M := by
+  rw [removeLoops, ← Loopless.ground_eq, restrict_ground_eq_self]
+
+lemma removeLoops_eq_self_iff : M.removeLoops = M ↔ M.Loopless := by
+  refine ⟨fun h ↦ ?_, fun h ↦ M.removeLoops_eq_self⟩
+  rw [← h]
+  infer_instance
+
+lemma removeLoops_isRestriction (M : Matroid α) : M.removeLoops ≤r M :=
+  restrict_isRestriction _ _ (fun _ h ↦ IsNonloop.mem_ground h)
+
+lemma eq_restrict_removeLoops (M : Matroid α) : M = M.removeLoops ↾ M.E := by
+  rw [removeLoops, ext_iff_indep]
+  simp only [restrict_ground_eq, restrict_indep_iff, true_and]
+  exact fun I hIE ↦ ⟨fun hI ↦ ⟨⟨hI,fun e heI ↦ hI.isNonloop_of_mem heI⟩, hIE⟩, fun hI ↦ hI.1.1⟩
+
+@[simp]
+lemma removeLoops_indep_eq : M.removeLoops.Indep = M.Indep := by
+  ext I
+  rw [removeLoops_eq_restrict, restrict_indep_iff, and_iff_left_iff_imp]
+  exact fun h e ↦ h.isNonloop_of_mem
+
+@[simp]
+lemma removeLoops_isBasis'_eq : M.removeLoops.IsBasis' = M.IsBasis' := by
+  ext
+  simp [IsBasis']
+
+@[simp] lemma removeLoops_isBase_eq : M.removeLoops.IsBase = M.IsBase := by
+  ext B
+  rw [isBase_iff_maximal_indep, removeLoops_indep_eq, isBase_iff_maximal_indep]
+
+@[simp]
+lemma removeLoops_isNonloop_iff : M.removeLoops.IsNonloop e ↔ M.IsNonloop e := by
+  rw [removeLoops_eq_restrict, restrict_isNonloop_iff, mem_setOf, and_self]
+
+lemma IsNonloop.removeLoops_isNonloop (he : M.IsNonloop e) : M.removeLoops.IsNonloop e :=
+  removeLoops_isNonloop_iff.2 he
+
+@[simp]
+lemma removeLoops_idem (M : Matroid α) : M.removeLoops.removeLoops = M.removeLoops := by
+  simp [removeLoops_eq_restrict]
+
+lemma removeLoops_restrict_eq_restrict (hX : X ⊆ {e | M.IsNonloop e}) :
+    M.removeLoops ↾ X = M ↾ X := by
+  rwa [removeLoops_eq_restrict, restrict_restrict_eq]
+
+@[simp]
+lemma restrict_univ_removeLoops_eq : (M ↾ univ).removeLoops = M.removeLoops := by
+  rw [removeLoops_eq_restrict, restrict_restrict_eq _ (subset_univ _), removeLoops_eq_restrict]
+  simp
+
+lemma removeLoops_loops_eq : M.removeLoops.loops = ∅ :=
+  Loopless.loops_eq_empty
```

Comment:
nitpick: this would make a bit more sense if it was proved directly after you show removeLoops is loopless. I'm also not entirely sure it needs to exist, but maybe it makes sense as a simp lemma?

Reply:
I think you're right that it doesn't need to exist - the linter won't let it be a simp lemma, because simp can prove it. Removed.